SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -10,36 +10,39 @@
 from xcpngutils.operationlocker import OperationLocker
 
 @error_wrapped
-def _list_disks():
-    disks = []
+def _list_devices():
+    devices = []
     result = run_command(['smartctl', '--scan'])
     for line in result['stdout'].splitlines():
-        if line.startswith('/dev/') and not line.startswith('/dev/bus/'):
-            disks.append(line.split()[0])
-    return disks
+        devices.append({'name': line.split()[0], 'type':line.split()[2]})
+    return devices
 
 @error_wrapped
 def get_information(session, args):
     results = {}
+    i = 0
     with OperationLocker():
-        disks = _list_disks()
-        for disk in disks:
-            cmd = run_command(["smartctl", "-j", "-a", disk], check=False)
-            results[disk] = json.loads(cmd['stdout'])
+        devices = _list_devices()
+        for device in devices:
+            cmd = run_command(["smartctl", "-j", "-a", "-d", devices[i]['name'], devices[i]['type']], check=False)
```

Comment:
You don't need the index anymore. Now you can use `device['name']` instead of `devices[i]['name']`. So you can remove the init `i = 0` and also the increment `i = i + 1`.